pkg/operator/controller/crl/controller.go

```diff
@@ -57,14 +58,27 @@ func New(mgr manager.Manager) (controller.Controller, error) {
 	}
 
 	// Index ingresscontrollers over the client CA configmap name so that
-	// configmapIsInUse and configmapToIngressController can look up
-	// ingresscontrollers that reference the configmap.
+	// clientCAConfigmapToIngressController can look up ingresscontrollers
+	// that reference the client CA configmap.
 	if err := operatorCache.IndexField(context.Background(), &operatorv1.IngressController{}, clientCAConfigmapIndexFieldName, client.IndexerFunc(func(o client.Object) []string {
 		ic := o.(*operatorv1.IngressController)
 		if len(ic.Spec.ClientTLS.ClientCA.Name) == 0 {
 			return []string{}
 		}
-		return []string{ic.Spec.ClientTLS.ClientCA.Name} // TODO Use the name of the operator-managed configmap.
+		return []string{operatorcontroller.ClientCAConfigMapName(ic).Name}
+	})); err != nil {
+		return nil, fmt.Errorf("failed to create index for ingresscontroller: %w", err)
+	}
+
+	// Index ingresscontrollers over the CRL configmap name so that
+	// crlConfigmapToIngressController can look up the ingresscontroller
+	// associated with a given CRL configmap.
+	if err := operatorCache.IndexField(context.Background(), &operatorv1.IngressController{}, crlConfigmapIndexFieldName, client.IndexerFunc(func(o client.Object) []string {
+		ic := o.(*operatorv1.IngressController)
+		if len(ic.Spec.ClientTLS.ClientCA.Name) == 0 {
```

Comment:
Do you still need to check the ClientCA.Name here?

Reply:
I think it's appropriate to do so.  The `len(ic.Spec.ClientTLS.ClientCA.Name) == 0` check is really intended to make sure we only index the ingresscontroller if it is using a client CA.  If there's no client CA, then there should be no CRL (the CRL is a list of certificates that were signed using the CA but have been since revoked).  I've added a short comment.  Does it all make sense?